src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,291 @@
+            {
+                throw new NotImplementedException();
+            }
+
+            var syntaxFactsService = document.GetLanguageService<ISyntaxFactsService>();
+
+            var lastUsageStatement = FindSiblingStatementContainingLastUsage(
+                declarationStatement,
+                localVariable,
+                semanticModel,
+                document.GetLanguageService<ISyntaxFactsService>(),
+                cancellationToken);
+
+            var parentStatementsList = GetStatements(parent);
+
+            var declarationStatementIndex = parentStatementsList.IndexOf(declarationStatement);
+
+            var lastUsageStatementIndex = lastUsageStatement is null
+                ? -1
+                : parentStatementsList.IndexOf(lastUsageStatement, declarationStatementIndex + 1);
+
+            var statementsToSurround = GetStatementsToSurround(parentStatementsList, declarationStatementIndex, lastUsageStatementIndex);
+
+
+            var trailingTrivia = SplitTrailingTrivia(declarationStatement, syntaxFactsService);
+
+            var usingStatement =
+                CreateUsingStatement(
+                    declarationStatement.WithoutTrivia(),
+                    trailingTrivia.sameLine,
+                    statementsToSurround)
+                    .WithLeadingTrivia(declarationStatement.GetLeadingTrivia())
+                    .WithTrailingTrivia(trailingTrivia.followingLines);
+
+            var newParent = WithStatements(
+                parent,
+                ReplaceRange(
+                    parentStatementsList,
+                    index: declarationStatementIndex,
+                    count: 1 + statementsToSurround.Count,
+                    newNodes: new[] { usingStatement }));
+
+            return document.WithSyntaxRoot(root.ReplaceNode(parent, newParent));
+        }
+
+        private static (SyntaxTriviaList sameLine, SyntaxTriviaList followingLines) SplitTrailingTrivia(SyntaxNode node, ISyntaxFactsService syntaxFactsService)
+        {
+            var trailingTrivia = node.GetTrailingTrivia();
+
+            for (var i = 0; i < trailingTrivia.Count; i++)
+            {
+                if (syntaxFactsService.IsEndOfLineTrivia(trailingTrivia[i]))
+                {
+                    return (
+                        sameLine: new SyntaxTriviaList(trailingTrivia.Take(i)),
+                        followingLines: new SyntaxTriviaList(trailingTrivia.Skip(i)));
+                }
+            }
+
+            return (
+                sameLine: trailingTrivia,
+                followingLines: SyntaxTriviaList.Empty);
+        }
+
+        protected static SyntaxList<TStatementSyntax> GetStatementsToSurround(SyntaxList<TStatementSyntax> parentList, int declarationStatementIndex, int lastUsageStatementIndex)
+        {
+            if (lastUsageStatementIndex == -1)
+            {
+                return default;
+            }
```

Comment:
i don't like this.

Reply:
What don't you like about it? Can't act on this feedback.